templates/adp-backend-template-dotnet/skeleton/test/${{ values.dotnet_solution_name }}.Core.Tests/ServicesTests/ItemServiceTests.cs

```diff
@@ -0,0 +1,118 @@
+using ${{ values.dotnet_solution_name }}.Core.Entities;
+using ${{ values.dotnet_solution_name }}.Core.Exceptions;
+using ${{ values.dotnet_solution_name }}.Core.Services;
+using AutoFixture;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+
+namespace ${{ values.dotnet_solution_name }}.Core.Tests.ServicesTests;
+
+[TestFixture]
+public class ItemServiceTests
+{
+    private readonly Fixture _fixture;
+    private readonly ILogger<ItemService> _mockLogger;
+    private ItemService _sut;
+
+    public ItemServiceTests()
+    {
+        _fixture = new Fixture();
+        _mockLogger = Substitute.For<ILogger<ItemService>>();
+    }
+
+    [SetUp]
+    public void SetUp()
+    {
+        _sut = new ItemService(_mockLogger);
+    }
+
+    [Test]
+    public async Task GetAllItems_ReturnsAllItems()
+    {
+        // Act
+        var result = await _sut.GetAllItems();
+        // Assert
+        result.Should().BeOfType<List<Item>>();
+        result.Should().HaveCountGreaterThan(0);
+    }
+
+    [Test]
+    public async Task GetById_Returns_Ok()
+    {
+        //Arrange
+        var item = _fixture.Create<Item>();
+        var createdItem = await _sut.CreateItem(item);
+        // Act
+        var result = await _sut.GetItemById(createdItem.Id);
+        // Assert
+        result.Should().BeOfType<Item>();
+        result.Should().NotBeNull();
+    }
+
+    [Test]
+    public void GetById_Returns_NotFound()
+    {
+        //Arrange
+        var id = 0;
+        // Act & Assert
+        Assert.ThrowsAsync<ItemNotFoundException>(async () => await _sut.GetItemById(id));
```

Comment:
Use fluent assertions for all assertions
```suggestion
        var test = () => _sut.GetItemById(id);
        await test.Should().ThrowAsync<ItemNotFoundException>()
```